### PR TITLE
Reconcile CNI in snow cluster reconciler

### PR DESCRIPTION
**Pull request by g-gaston**
*Testing (if applicable):*
Unit tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: controllers/factory.go [hunk 1]
```diff
@@ -213,14 +213,18 @@ func (f *Factory) withVSphereClusterReconciler() *Factory {
 }
 
 func (f *Factory) withSnowClusterReconciler() *Factory {
-	f.withCNIReconciler()
+	f.withCNIReconciler().withTracker()
```
Comment (vivek-koppuru) [inline controllers/factory.go:216]: What is this?

Comment (g-gaston) [reply controllers/factory.go:216]: tracker? You can see it here https://github.com/aws/eks-anywhere/blob/079ca2f5b838e81b9f7446519b0e895cc3205bf1/controllers/factory.go#L132-L155

It builds a remote client tracker. It builds kube clients for workload clusters from kubeconfigs stored in secrets.

**Review by mitalipaygude — APPROVED** [review]

File: pkg/providers/snow/reconciler/reconciler.go [hunk 2]
```diff
@@ -75,11 +104,10 @@ func (s *Reconciler) ReconcileWorkers(ctx context.Context, log logr.Logger, clus
...
 }
 
-func (s *Reconciler) ReconcileControlPlane(ctx context.Context, log logr.Logger, clusterSpec *cluster.Spec) (controller.Result, error) {
-	log = log.WithValues("phase", "reconcileControlPlane")
-	log.Info("Applying control plane CAPI objects")
-
-	return s.Apply(ctx, func() ([]kubernetes.Object, error) {
-		return snow.ControlPlaneObjects(ctx, clusterSpec, clientutil.NewKubeClient(s.client))
-	})
+func capiClusterObjectKey(cluster *anywherev1.Cluster) client.ObjectKey {
```
Comment (taneyland) [inline pkg/providers/snow/reconciler/reconciler.go:107]: Should we also define this in one of the capi files that you are introducing in #3193? 
Maybe in `pkg/controller/clusters/clusterapi.go`

Comment (g-gaston) [reply pkg/providers/snow/reconciler/reconciler.go:107]: we probably should yeah
what about we make this a follow up? I don't have time right now to move it and I want to start the testing as soon as possible. Or maybe you can move the method and add a test? And just push to this branch :)

Comment (g-gaston) [reply pkg/providers/snow/reconciler/reconciler.go:107]: Talked about it offline, we will have a follow up PR

**Pull request merged**